1-JavaScript/Week3/homework/ex7-mindPrivacy.js

```diff
@@ -26,8 +26,16 @@ const employeeRecords = [
   },
 ];
 
-function filterPrivateData(/* parameter(s) go here */) {
-  // TODO complete this function
+function filterPrivateData(arrayEmployee) {
+
+  const myArray = [];
+
+  for (const {name, occupation, email} of arrayEmployee){
+  const myObject = {name, occupation, email};
+  myArray.push(myObject)
+  }
+
+return myArray
```

Comment:
I'm impressed that this works! It's a little bit of an odd solution, but good job figuring it out, especially with the creation of a new const using a destructor in the for loop, intersting!

I do have a suggestion that might be a bit cleaner:

```suggestion
function filterPrivateData(arrayEmployees) {
  return arrayEmployees.map((employee) => {
    const {name, occupation, email} = employee
    return {name, occupation, email}
  })
}
```

Notice that there is more separation of logic; there is the "map" which doe just that, it maps, then there is the destructing line, and then there is the returning of the new object.

Reply:
That is so cool and hard to understand for me as a beginner. :)

Reply:
Oh really? to me this code looks easier to read than what you had! But either version works fine, so you can pick whichever you prefer in this case:)